xcube/core/store/store.py

```diff
@@ -690,3 +691,22 @@ def deregister_data(self, data_id: str):
         Raises:
             DataStoreError: If an error occurs.
         """
+
+
+class PreloadDataStore(MutableDataStore):
+    """A preload data store is a multable data store which contains the preload handle.
+
+    Instances of this class are returned by the ``DataStore.preload_data()`` method.
+    """
+
+    @property
+    @abstractmethod
+    def preload_handle(self) -> PreloadHandle:
+        """This field must be defined and set in subclasses."""
```

Comment:
Please explain the property's purpose rather than saying that the property is abstract - that's already indicated by the `@abstractmethod` decorator. You can tell developers what is expected here.

```suggestion
        """The preload handle that can be used to observe the preload progress.
        Implementors of this interface may use a `DefaultProgressHandle` or consider
        returning a `NullProgressHandle` if the progress is not observable.        
        """
```

Reply:
changes slightly as the classes are called NullPreloadHandle and ExecutorPreloadHandle in the preload module.